src/main/java/edu/umich/its/spe/SPEMaster.java

```diff
@@ -0,0 +1,378 @@
+// TODO: test error handling
+
+// Spring: make visible for auto wiring.
+@Component
+public class SPEMaster {
+
+	static final Logger M_log = LoggerFactory.getLogger(SPEMaster.class);
+	static HashMap<String,String> unirest;
+
+	// Let Spring inject the properties, the esb service, and the persistString implementation.
+	@Autowired
+	private SPEEsb speesb;
+
+	@Autowired
+	private PersistString persistString;
+
+	@Autowired
+	private SPEProperties speproperties;
+
+	// If no other timestamp is specified use the current time.
+	private LocalDateTime startingTime = LocalDateTime.now();
+	
+	public SPEMaster() {
+		super();
+		M_log.info("SPEMaster: this: "+this.toString());
+	}
+
+	// Set global timeouts for EBS calls 
+	protected void setUnirestGlobalValues() {
+		
+		M_log.info("setupUnirestGlobalValues");
+		unirest = speproperties.getUnirest();
+		M_log.info("unirest properties: {}",unirest);
+		
+		// Setup values for request timeouts.
+		long ct = longFromStringWithDefault(unirest.get("connectionTimeout"), 10000l);
+		long st = longFromStringWithDefault(unirest.get("socketTimeout"), 10000l);
+		
+		M_log.info("unirest timeouts: connectionTimeout: {} socketTimeout: {}",ct,st);
+		
+		Unirest.setTimeouts(ct,st);
+	}
+
+	// Get a long value from a string use default if the string is null or empty.
+	protected long longFromStringWithDefault(String longString, Long defaultValue) {
+		long longValue = (longString == null ? defaultValue : Long.parseLong(longString));
+		M_log.info("longFromString: {} {}",longString,longValue);
+		return longValue;
+	}
+
+	/*********************** Orchestrate the script. *****************/ 
+	
+	/* 
+	 * - configuration is read and injected by spring
+	 * - get value gradeAfterTime from persist string (default if necessary).
+	 * - get grades.
+	 * - translate to format for updates.
+	 * - send to MPathways.
+	 * 
+	 */
+	
+	/* Organize the task and handle errors */
+	
+	public void orchestrator () throws PersistStringException, SPEEsbException {
+	
+		M_log.info("Start SPE orchestrator");
+	
+		setUnirestGlobalValues();
+		
+		// sanity check by renewing token to ensure we can connect to the database.
+		
+		if (!verifyESB()) {
+			M_log.error("Unable to connect to esb"); 
+			throw new SPEEsbException("Unable to connect to ESB");
+		}
+		
+		//// Get the relevant grades.
+		String lastUpdateTime = readLastGradeTransferTime();
+		M_log.error("fake lastUpdateTime");
+		lastUpdateTime = "2017-03-08 15:38:21";
+		M_log.info("lastUpdateTime: {}",lastUpdateTime);
+		String assignmentsFromDW;
+	
+		try {
+			assignmentsFromDW = getSPEGrades(lastUpdateTime);
+	
+			//// Extract grades from JSON the grades for insertion	
+	
+			ArrayList<HashMap<String,String>> SPEgradeMaps = null;
+			try {
+				SPEgradeMaps = convertSPEGradesFromDataWarehouseJSON(assignmentsFromDW);
+			} catch (JSONException e) {
+				M_log.error("exception in converting grades: "+e);
+			}
+	
```

Comment:
If an exception happen then the flow should return from here, but as of not there is a chance to go to next statement here. Either Throw and exception or add a return statement catch block